src/toil_scripts/lib/programs.py

```diff
@@ -83,21 +101,53 @@ def docker_call(tool,
     if env:
         for e, v in env.iteritems():
             base_docker_call.extend(['-e', '{}={}'.format(e, v)])
+    
     if docker_parameters:
         base_docker_call += docker_parameters
-
-    _log.debug("Calling docker with %s." % " ".join(base_docker_call + [tool] + parameters))
-
-    docker_call = base_docker_call + [tool] + parameters
-
+   
+    docker_call = []
+    
+    # Pipe functionality
+    #   each element in the parameters list must represent a sub-pipe command
+    shell_flag = False      # Flag for running subprocess with string command or list command
+    if isinstance(tool, list):
+        command_list = []
+        if len(tool) > 1:
+            # If tool is a list containing multiple docker container name strings
+            #   then format the docker call in the 'pipe-of-containers' mode
+            docker_call.extend(base_docker_call + ['--entrypoint /bin/bash', tool[0], '-c \'{}\''.format(parameters[0])])
+            for i in xrange(1, len(tool)):
+                docker_call.extend(['|'] + base_docker_call + ['-i --entrypoint /bin/bash', tool[i], '-c \'{}\''.format(parameters[i])])
+            docker_call = " ".join(docker_call)
+            shell_flag=True
+            print("Calling docker with {}".format(docker_call)) 
+            _log.debug("Calling docker with %s." % docker_call)
+            
+        elif len(tool) == 1:
+            # If tool is a list containing a single docker container name string
+            #   then format the docker call in the 'pipe-in-single-container' mode
+            docker_call.extend(base_docker_call + ['--entrypoint /bin/bash', tool[0], '-c \'{}\''.format(" | ".join(parameters))])
+            docker_call = " ".join(docker_call)
+            shell_flag=True
+            print("Calling docker with {}".format(docker_call)) 
+            _log.debug("Calling docker with %s." % docker_call)
+             
+    else:        
+        docker_call = base_docker_call + [tool] + parameters
+        _log.debug("Calling docker with %s." % " ".join(docker_call))
+    
     try:
         if outfile:
-            subprocess.check_call(docker_call, stdout=outfile)
+            subprocess.check_call(docker_call, stdout=outfile, shell=shell_flag)
         else:
             if check_output:
```

Comment:
So essentially can you make it so that `check_output=True` yields only stdout, `return_stderr=True` yields only stderr, while `check_output=True, return_stderr=True` yields both?